dandiapi/api/views/asset.py

```diff
@@ -85,7 +86,10 @@ def create(self, request, version__dandiset__pk, version__version):
             metadata=asset_metadata,
             version=version,
         )
-        asset.save()
+        try:
+            asset.save()
+        except IntegrityError:
+            return Response('asset already exists', status=status.HTTP_400_BAD_REQUEST)
```

Comment:
is that the only `IntegrityError` case whenever it could be raised?  may be so ATM, but would it be always the only one? ;)
I think it would be quite bad if error message is not factually correct...
from https://github.com/dandi/dandi-api/issues/65 it looks like 
```
django.db.utils.IntegrityError: duplicate key               
Jan 25 15:15:40 dandi-api app/web.1                     value violates unique constraint                            
Jan 25 15:15:40 dandi-api app/web.1                     "api_asset_path_version_id_0267c3d9_uniq"                   
Jan 25 15:15:40 dandi-api app/web.1                     DETAIL:  Key (path,                                         
Jan 25 15:15:40 dandi-api app/web.1                     version_id)=(sub-RAT123/sub-RAT123.nwb, 10)                 
Jan 25 15:15:40 dandi-api app/web.1                     already exists
```

I hate doing matching in outputs or messages, but may be first match for `duplicate key  value violates unique constraint api_asset_path_version_id` and if that matches -- return the above response, but ideally include that path into it (could help debugging while trying to upload many files and some file failing).  If no match -- just a `str` of that exception?

Reply:
re
> return the above response

I would also improve that as `f'there is already an asset at {path}'` or alike.  "asset already exists" is a bit of, since in my mind assets are uniq, and I am just creating a new asset, so how could it already "exist"? ;)

Reply:
Turns out there is a semi-formal way to accomplish this: https://stackoverflow.com/questions/25368020/django-deduce-duplicate-key-exception-from-integrityerror